components/board/genericlinux/board.go

```diff
@@ -205,28 +209,96 @@ func (b *sysfsBoard) reconfigureAnalogs(ctx context.Context, newConf *Config) er
 	return nil
 }
 
-func (b *sysfsBoard) reconfigureGpios(newConf *Config) error {
+func (b *sysfsBoard) reconfigureInterrupts(newConf *Config) error {
 	if b.usePeriphGpio {
 		if len(newConf.DigitalInterrupts) != 0 {
 			return errors.New("digital interrupts on Periph GPIO pins are not yet supported")
 		}
 		return nil // No digital interrupts to reconfigure.
 	}
 
-	// TODO(RSDK-2684): we dont configure pins so we just unset them here. not really great behavior.
-	// We currently have two implementations of GPIO pins on these boards: one using
-	// libraries from periph.io and one using an ioctl approach. If we're using the
-	// latter, we need to initialize it here.
-	gpios, interrupts, err := b.gpioInitialize( // Defined in gpio.go
-		b.cancelCtx,
-		b.gpioMappings,
-		newConf.DigitalInterrupts,
-		b.logger)
-	if err != nil {
-		return err
+	// If we get here, we need to reconfigure b.interrupts. Any pin that already exists in the
+	// right configuration should just be copied over; closing and re-opening it risks losing its
+	// state.
+	newInterrupts := make(map[string]*digitalInterrupt, len(newConf.DigitalInterrupts))
+
+	// This helper function finds the new config (if any) for a pre-existing digital interrupt.
+	findNewDigIntConfig := func(interrupt *digitalInterrupt) *board.DigitalInterruptConfig {
```

Comment:
[if-minor] the outer function is getting quite long, how would you feel about moving this function out to the file and having it take an additional argument of type `[]board.DigitalInterruptConfig`?